packages/ra-core/src/dataProvider/useUpdate.ts

```diff
@@ -296,12 +327,17 @@ export const useUpdate = <RecordType extends RaRecord = any, ErrorType = Error>(
         const {
             mutationMode,
             returnPromise = mutationOptions.returnPromise,
+            onError,
+            onSettled,
```

Comment:
I struggle to understand how onSuccess is handled. I think your code is correct, but as onSuccess is buried in otherCallTimeOptions, it's hard to make sure it's called at the right place. I think the code would be more legible if you extracted here, and passed it explicitly in each mutation mode.